src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -3,45 +3,195 @@
 import java.util.List;
 
 public class MyLinkedList<T> implements MyLinkedListInterface<T> {
+    private Node<T> head;
+    private Node<T> tail;
+    private int size = 0;
+
+    public static class Node<E> {
+        private E item;
+        private Node<E> prev;
+        private Node<E> next;
+
+        public Node(Node<E> prev, E item, Node<E> next) {
+            this.item = item;
+            this.prev = prev;
+            this.next = next;
+        }
+    }
+
+    private void checkIndex(int someIndex) {
+        if (someIndex < 0 || someIndex >= size) {
+            throw new IndexOutOfBoundsException("Not appropriate index.");
+        }
+    }
+
     @Override
     public void add(T value) {
+        Node<T> node;
+        node = new Node<>(tail, value, null);
+
+        if (size == 0) {
+            head = node;
+        } else {
+            tail.next = node;
+        }
+
+        tail = node;
+        size++;
     }
 
     @Override
     public void add(T value, int index) {
+        if (index == size) {
+            add(value);
+
+            return;
+        }
+
+        if (index == 0) {
+            Node<T> node = new Node<>(null, value, head);
+            head.prev = node;
+            head = node;
+
+            size++;
+
+            return;
+        }
```

Comment:
```suggestion
        if (index == size) {
            add(value);
            return;
        }

        if (index == 0) {
            Node<T> node = new Node<>(null, value, head);
            head.prev = node;
            head = node;
            size++;
            return;
        }
```
в джаві не ставиться так багато відступів як у JS 🙂
Тільки там, де потірбно відділити логіку (наприклад if)

Reply:
done :)

Reply:
а перед return теж відступ не потрібен?

Reply:
ні)